pallets/loans/src/lib.rs

```diff
@@ -770,6 +765,174 @@ pub mod pallet {
 
 			Ok(Some(T::WeightInfo::update_portfolio_valuation(count)).into())
 		}
+
+		/// Transfer debt from one loan to another loan,
+		/// repaying from the first loan and borrowing the same amount from the
+		/// second loan. `from_loan_id` is the loan used to repay.
+		/// `to_loan_id` is the loan used to borrow.
+		/// The repaid and borrow amount must match.
+		#[pallet::weight(T::WeightInfo::propose_transfer_debt(T::MaxActiveLoansPerPool::get()))]
+		#[pallet::call_index(11)]
+		pub fn propose_transfer_debt(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			from_loan_id: T::LoanId,
+			to_loan_id: T::LoanId,
+			repaid_amount: RepaidPricingAmount<T>,
+			borrow_amount: PricingAmount<T>,
```

Comment:
I think this naming is confusing. Generally pricing refers to setting loan parameters (e.g. updating the interest rate).

What about `BorrowInput` and `RepayInput`?

Reply:
These "Pricing" names come because the `pricing` module defines them as follows:
```rust
pub enum PricingAmount<T: Config> {
	Internal(T::Balance),
	External(external::ExternalAmount<T>),
}
```
So, it carries the "pricing mode" information in the struct.

But it's also true that these structs are also used for introducing values into the system and are never stored, so I think `Input` is a good name for those too 👍🏻

Reply:
But the information included in that `PricingAmount` struct is not pricing information. It's just that the borrow/repay input is dependent on the pricing type. I would even say the `PricingAmount` struct semantically should not be in the `pricing.rs` file for that reason.

Reply:
Mmm... good point. I'll rethink if it should be moved to a better place. And definitely `Input` suffix fits here 👍🏻

Reply:
Ok, looking deeply into it, I think the problem here is that `RepaidPricingAmount` contains into it a `PricingAmount`, so we can not change them to `BorrowInput/RepayInput` names. That would mean that for composing a `RepaidInput`, the user needs to create a `BorrowInput`.

I like the `Input` suffix, so I'd opt for these new names:

- `PrincipalInput`
- `RepaidInput` (which is composed using a `PricipalInput`).

And those types can be moved to a new module `/entities/input.rs`

WDYT?

Reply:
Yeah that's a great solution 💯